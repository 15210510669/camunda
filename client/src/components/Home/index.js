/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {default as Home} from './Home';
export {default as Collection} from './Collection';
