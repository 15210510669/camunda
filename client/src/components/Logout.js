/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {useEffect} from 'react';
import {withRouter} from 'react-router-dom';
import {get} from 'request';
import {withErrorHandling} from 'HOC';
import {addNotification} from 'notifications';
import {t} from 'translation';

export function Logout({mightFail, history}) {
  useEffect(() => {
    mightFail(
      get('api/authentication/logout'),
      () => {
        addNotification({text: t('navigation.logoutSuccess')});
        history.replace('/');
      },
      () => {
        addNotification({text: t('navigation.logoutFailed'), type: 'error'});
        history.replace('/');
      }
    );
  }, [mightFail, history]);

  return null;
}

export default withRouter(withErrorHandling(Logout));
