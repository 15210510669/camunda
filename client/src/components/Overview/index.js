export {default as Overview} from './Overview';
export {default as Alerts} from './Alerts';
