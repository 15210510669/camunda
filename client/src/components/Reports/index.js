export {default as Reports} from './Reports';
export {default as Report} from './Report';
export {Filter} from './filter'
