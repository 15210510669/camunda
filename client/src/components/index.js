export {Footer} from './Footer';
export {Header, LogoutButton} from './Header';
export {Home} from './Home';
export {Login} from './Login';
export {PrivateRoute} from './PrivateRoute';
export {Dashboard} from './Dashboard';
export {Dashboards} from './Dashboards';
export {Report} from './Report';
export {Reports} from './Reports';
