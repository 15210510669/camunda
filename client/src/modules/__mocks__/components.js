export {default as Button} from '../components/__mocks__/Button';
export {default as Input} from '../components/__mocks__/Input';
export {default as Modal} from '../components/__mocks__/Modal';
export {default as Popover} from '../components/__mocks__/Popover';
export {default as ControlGroup} from '../components/__mocks__/ControlGroup';
export {default as ShareEntity} from '../components/__mocks__/ShareEntity';
export {default as DashboardView} from '../components/__mocks__/DashboardView';
export {default as Icon} from '../components/__mocks__/Icon';
export {default as Dropdown} from '../components/__mocks__/Dropdown';
export {default as Message} from '../components/__mocks__/Message';
export {default as ReportView} from '../components/__mocks__/ReportView';
export {default as Select} from '../components/__mocks__/Select';
export {default as BPMNDiagram} from '../components/__mocks__/BPMNDiagram';
export {default as TargetValueBadge} from '../components/__mocks__/TargetValueBadge';
export {default as Table} from '../components/__mocks__/Table';
export {default as Logo} from '../components/__mocks__/Logo';
export {default as ErrorMessage} from '../components/__mocks__/ErrorMessage';
