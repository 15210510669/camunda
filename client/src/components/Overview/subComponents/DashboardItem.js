import React from 'react';
import {Button, Icon} from 'components';
import LastModified from './LastModified';
import {Link} from 'react-router-dom';
import entityIcons from '../entityIcons';
import CollectionsDropdown from './CollectionsDropdown';
import {formatters} from 'services';

const EntityIcon = entityIcons.dashboard.generic.Component;

export default function DashboardItem({
  dashboard,
  duplicateEntity,
  showDeleteModalFor,
  collection,
  searchQuery
}) {
  return (
    <li className="DashboardItem listItem">
      <Link className="info" to={`/dashboard/${dashboard.id}`}>
        <span className="icon">
          <EntityIcon />
        </span>
        <div className="textInfo">
          <div className="data dataTitle">
            <h3>{formatters.getHighlightedText(dashboard.name, searchQuery)}</h3>
          </div>
          <div className="extraInfo">
            <span className="data custom">
              {dashboard.reports.length} Report
              {dashboard.reports.length !== 1 ? 's' : ''}
            </span>
            <LastModified
              label="Last changed"
              date={dashboard.lastModified}
              author={dashboard.lastModifier}
            />
          </div>
        </div>
      </Link>
      <CollectionsDropdown entity={dashboard} currentCollection={collection} />
      <div className="operations">
        <Link title="Edit Dashboard" to={`/dashboard/${dashboard.id}/edit`}>
          <Icon title="Edit Dashboard" type="edit" className="editLink" />
        </Link>
        <Button
          title="Duplicate Dashboard"
          onClick={duplicateEntity('dashboard', dashboard, collection)}
        >
          <Icon type="copy-document" title="Duplicate Dashboard" className="duplicateIcon" />
        </Button>
        <Button
          title="Delete Dashboard"
          onClick={showDeleteModalFor({
            type: 'dashboard',
            entity: dashboard
          })}
        >
          <Icon type="delete" title="Delete Dashboard" className="deleteIcon" />
        </Button>
      </div>
    </li>
  );
}
