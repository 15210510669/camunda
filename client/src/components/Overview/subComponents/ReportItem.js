import React from 'react';
import {Button, Icon} from 'components';
import {getReportInfo, getReportIcon} from '../service';
import LastModified from './LastModified';
import {Link} from 'react-router-dom';
import CollectionsDropdown from './CollectionsDropdown';
import {formatters} from 'services';

export default function ReportItem({
  report,
  searchQuery,
  duplicateEntity,
  showDeleteModalFor,
  collection
}) {
  const {Icon: ReportIcon, label} = getReportIcon(report);

  return (
    <li className="ReportItem listItem">
      <Link className="info" to={`/report/${report.id}`}>
        <span className="icon" title={label}>
          <ReportIcon />
        </span>
        <div className="textInfo">
          <div className="data dataTitle">
            <h3>{formatters.getHighlightedText(report.name, searchQuery)}</h3>
            {report.combined && <span>Combined</span>}
            {report.reportType && report.reportType === 'decision' && <span>Decision</span>}
          </div>
          <div className="extraInfo">
            <span className="data custom">{getReportInfo(report)}</span>
            <LastModified
              label="Last changed"
              date={report.lastModified}
              author={report.lastModifier}
            />
          </div>
        </div>
      </Link>
      <CollectionsDropdown entity={report} currentCollection={collection} />
      <div className="operations">
        <Link title="Edit Report" to={`/report/${report.id}/edit`}>
          <Icon title="Edit Report" type="edit" className="editLink" />
        </Link>
        <Button title="Duplicate Report" onClick={duplicateEntity('report', report, collection)}>
          <Icon type="copy-document" title="Duplicate Report" className="duplicateIcon" />
        </Button>
        <Button
          title="Delete Report"
          onClick={showDeleteModalFor({type: 'report', entity: report})}
        >
          <Icon type="delete" title="Delete Report" className="deleteIcon" />
        </Button>
      </div>
    </li>
  );
}
