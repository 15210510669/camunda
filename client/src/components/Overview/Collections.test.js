/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import {Button} from 'components';

import CollectionsWithStore from './Collections';
import ReportItem from './subComponents/ReportItem';
import DashboardItem from './subComponents/DashboardItem';
const Collections = CollectionsWithStore.WrappedComponent;

const processReport = {
  id: 'reportID',
  name: 'Some Report',
  lastModifier: 'Admin',
  lastModified: '2017-11-11T11:11:11.1111+0200',
  reportType: 'process',
  combined: false
};

const dashboard = {
  id: 'dashboardID',
  name: 'Some Dashboard',
  lastModifier: 'Admin',
  lastModified: '2017-11-11T11:11:11.1111+0200',
  reports: []
};

const collection = {
  id: 'aCollectionId',
  name: 'aCollectionName',
  created: '2017-11-11T11:11:11.1111+0200',
  owner: 'user_id',
  data: {
    configuration: {},
    entities: [processReport, dashboard]
  }
};

const collectionWithManyReports = {
  id: 'aCollectionId',
  name: 'aCollectionName',
  created: '2017-11-11T11:11:11.1111+0200',
  owner: 'user_id',
  data: {
    configuration: {},
    entities: new Array(6).fill(processReport)
  }
};

const props = {
  store: {
    updating: null,
    collections: [collection],
    searchQuery: ''
  }
};

it('should show no data indicator', () => {
  const node = shallow(<Collections {...props} store={{collections: [], searchQuery: ''}} />);

  expect(node.find('.collectionBlankSlate')).toBePresent();
});

it('should show the list of entities when entity has open state', () => {
  const node = shallow(<Collections {...props} />);
  node.setState({[collection.id]: true});

  expect(node.find('.entityList')).toBePresent();
});

it('should not show a button to show all reports if the number of reports is less than 5', () => {
  const node = shallow(<Collections {...props} />);

  node.setState({[collection.id]: true});

  expect(node).not.toIncludeText('Show all');
});

it('should show a button to show all reports if the number of reports is greater than 5', () => {
  const store = {collections: [collectionWithManyReports], searchQuery: ''};
  const node = shallow(<Collections {...props} store={store} />);

  node.setState({[collection.id]: true});

  expect(node.find(Button)).toIncludeText('Show all');
});

it('should show a button to show less reports if the number of reports is greater than 5', () => {
  const store = {collections: [collectionWithManyReports], searchQuery: ''};
  const node = shallow(<Collections {...props} store={store} />);
  node.setState({[collection.id]: true});

  node.find(Button).simulate('click');

  expect(node.find(Button)).toIncludeText('Show less...');
});

it('should render dashboard and report list items', () => {
  const node = shallow(<Collections {...props} />);
  node.setState({[collection.id]: true});

  expect(node.find(ReportItem)).toBePresent();
  expect(node.find(DashboardItem)).toBePresent();
});

it('should show no result found text when no matching reports were found', () => {
  const node = shallow(<Collections {...props} store={{...props.store, searchQuery: 'test'}} />);

  expect(node.find('.empty')).toMatchSnapshot();
});
