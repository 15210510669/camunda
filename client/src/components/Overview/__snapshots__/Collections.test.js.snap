// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show no result found text when no matching reports were found 1`] = `
<p
  className="empty"
>
  No Collections matching '
  test
  '
</p>
`;
